### PR TITLE
Prevent offline dinosaur

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
If offline, either show cached content or a special offline page (`/.shell/offline`).

### CODE REVIEW

File: public/sw.js [hunk 1]
```diff
@@ -1,15 +1,30 @@
 /* eslint-env serviceworker, browser */
 
-importScripts('/sw-toolbox/sw-toolbox.js');
+importScripts('/sw-toolbox/sw-toolbox.js'); /* global toolbox */
 
-self.addEventListener('install', () => {
-  console.log('INSTALL');
-});
+// The offline page itself *and* its dependencies
```
Comment (ithinkihaveacat) [inline public/sw.js]: This is annoying. Don't see any easy way to generate this. (Annoying that `/.shell/offline` needs to be a real endpoint, too.)

**Comment by juliantoledo** [comment]
@ithinkihaveacat Nice progress with offline!
I think we should disable the login buttons.

Force reload shows the offline dinosaur, if I reload again it shows the page. Is that normal for sw?

I also found that if the DB is offline the page hangs on the server, we should do something about that in another PR, maybe let the sw kick in after a timeout.

**Comment by andreban** [comment]
I agree with disabling login buttons.

@juliantoledo force-reload showing the offline dinosaur is expected. It makes the browser skip the service-worker. Regular reload should work.

It's possible to tweak timeouts in 'sw-toolbox'. I think it's important in order to solve the LieFi scenario. But agreed we can file an issue and do it on a following PR.

**Comment by juliantoledo** [comment]
LGTM, it's working really good now.

I will open a feature request to grey out PWAs that are not available offline.

**Pull request merged**